test/Reentrancy/GMXProtocol/ExploitSC.sol

```diff
@@ -0,0 +1,584 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import "forge-std/Test.sol";
+import {TestHarness} from "../../TestHarness.sol";
+import {TokenBalanceTracker} from "../../modules/TokenBalanceTracker.sol";
+import {IVault, IVaultUtils} from "./IVault.sol";
+import {IOrderBook} from "./IOrderBook.sol";
+import {IRouter} from "./IRouter.sol";
+import {IWETH9} from "../../interfaces/IWETH9.sol";
+import {IUniswapV3Pair} from "../../utils/IUniswapV3Pair.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IRewardRouterV2} from "./IRewardRouterV2.sol";
+import {IGlpManager} from "./IGlpManager.sol";
+
+import {IPositionRouter} from "./IPositionRouter.sol";
+
+// Attacker's Smart Contract 1
+contract ExploitSC is TestHarness, TokenBalanceTracker {
```

Comment:
we could significantly improve this reproduction by splitting / dividing the exploit into sections. (e.g. section 1: group all approvals, section 2: trigger the attack, etc) logging each section as a way to show a timeline.